sparkler-plugins/fetcher-htmlunit/src/main/java/edu/usc/irds/sparkler/plugin/HtmlUnitFetcher.java

```diff
@@ -0,0 +1,147 @@
+
+    private static final Integer DEFAULT_TIMEOUT = 2000;
+    private static final Integer DEFAULT_JS_TIMEOUT = 2000;
+    private static final Logger LOG = LoggerFactory.getLogger(HtmlUnitFetcher.class);
+    private WebClient driver;
+
+    @Override
+    public void init(JobContext context, String pluginId) throws SparklerException {
+        super.init(context, pluginId);
+        //TODO: get timeouts from configurations
+        driver = new WebClient(BrowserVersion.BEST_SUPPORTED);
+        driver.setJavaScriptTimeout(DEFAULT_JS_TIMEOUT);
+        WebClientOptions options = driver.getOptions();
+        options.setCssEnabled(false);
+        options.setAppletEnabled(false);
+        options.setDownloadImages(false);
+        options.setJavaScriptEnabled(true);
+        options.setTimeout(DEFAULT_TIMEOUT);
+        options.setUseInsecureSSL(true);
+        options.setPopupBlockerEnabled(true);
+        options.setDoNotTrackEnabled(true);
+        options.setGeolocationEnabled(false);
+        options.setHistorySizeLimit(2);
+        options.setPrintContentOnFailingStatusCode(false);
+        options.setThrowExceptionOnScriptError(false);
+        options.setThrowExceptionOnFailingStatusCode(false);
+        if (this.httpHeaders != null && !this.httpHeaders.isEmpty()) {
+            LOG.info("Found {} headers", this.httpHeaders.size());
+            this.httpHeaders.forEach((name, val) -> driver.addRequestHeader(name, val));
+        } else {
+            LOG.info("No user headers found");
+        }
+    }
+
+    @Override
+    public FetchedData fetch(Resource resource) throws Exception {
+        LOG.info("HtmlUnit FETCHER {}", resource.getUrl());
+        FetchedData fetchedData;
+        try {
+            String userAgent = getUserAgent();
+            if (StringUtils.isNotBlank(userAgent)) {
+                driver.removeRequestHeader(USER_AGENT);
+                driver.addRequestHeader(USER_AGENT, userAgent);
+            }
+            Page page = driver.getPage(resource.getUrl());
+
+            WebResponse response = page.getWebResponse();
+            boolean truncated = false;
+            try (InputStream stream = response.getContentAsStream()) {
+                try (BoundedInputStream boundedStream = new BoundedInputStream(stream, CONTENT_LIMIT)) {
+                    try (ByteArrayOutputStream out = new ByteArrayOutputStream()) {
+                        IOUtils.copy(boundedStream, out);
+                        fetchedData = new FetchedData(out.toByteArray(), response.getContentType(), response.getStatusCode());
+                        long contentLength = page.getWebResponse().getContentLength();
+                        if (contentLength > 0 && contentLength < Integer.MAX_VALUE) {
+                            fetchedData.setContentLength((int) contentLength);
+                            truncated = (contentLength > fetchedData.getContentLength());
```

Comment:
We are including TRUNCATED flag to headers below.
```
if (truncated){ 
    headers.put(TRUNCATED, Collections.singletonList(Boolean.TRUE.toString()))
}
```
I agree we should also add LOG statement here. I will fix this too